Step By Step Guide To Klasa (#705)
dirigeants/klasa: guides/Step By Step Guide/CustomizingTheClient.md

Congrats! If you reached this part of the guide, you have successfully created your very own Discord Bot! Now, its time to understand the options Klasa gives us so we can customize the bot.

> Note: This guide was intended for beginner developers so we will go step by step. However, if you wish to speed things up you can use this code snippet below so you don't have to keep restarting the bot. Otherwise, just follow along with the guide to understand how each of these options work.

```js
const BotClient = new Client({
	commandEditing: true,
	commandLogging: true,
	prefix: `+`,
	readyMessage: (client) => `${client.user.tag}, ready to serve ${client.guilds.size} guilds.`
});
```

## Adding Your Own Default Prefix

> Remember how we used `@bot help` to get the bot to send us the help command. The reason `@bot` worked is because Klasa adds the bots @ as a default prefix.

Reviewer: ```suggestion
Remember how we used `@bot help` to get the bot to send us the help command. The reason `@bot` worked is because Klasa adds the bots mention tag as a default prefix.
```

> Note: If you don't like this behavior you can remove it, since Klasa is extremely customizable, but let's leave it in for now. You can review the Advanced section of the guide to learn how.

Suppose we want the bot to respond to `+` as the prefix instead of @. To do this, we go back into the `index` file we made before and add in the prefix we want. You can choose whichever prefix you wish, but for this guide we will be using `+`.

> > Note: It is important to remember that this option here is a DEFAULT prefix. This means that this prefix can NOT be customized by servers. This is a prefix that you can have users use when they forget their own custom server prefix.

Reviewer: Uh... I'm lost here, what does this mean? It's customizable per-guild basis.

```js
const BotClient = new Client({
	prefix: `+`
}).login(`your-bot-token`);
```

> > IMPORTANT: We have to restart the bot to make this change since this index file is NOT a Klasa Piece. We will learn more about Klasa Pieces and how easy it is to edit them without restarting the bot when we create our first piece. For now, just remember editing this index file will always require the bot to be restarted.

Reviewer: Okay and... what's a "Klasa Piece"? If this is going to be step-by-step, then you should either mention it before, or put a link to the title (`{@scrollto title}`) or guide (`{@tutorial name}`).

Klasa made us an amazing `reboot` command so go ahead and type `@bot reboot`
<!-- Insert Image Here -->

> > Note: If you are having trouble with the Reboot command, please see the FAQ.

Reviewer: ```suggestion
> Note: If you are having trouble with the Reboot command, please see the {@tutorial FAQ}.
```

<!-- Insert Image Below -->

## Enabling Command Editing

One last thing we can cover in this guide is `Command Editing`. This feature allows users to simply edit their message to run the command again. The best way to understand it, is to see it in action. Let's start by enabling this option:

```js
const BotClient = new Client({
	commandEditing: true,
	commandLogging: true,
	prefix: `+`,
	readyMessage: (client) => `${client.user.tag}, ready to serve ${client.guilds.size} guilds.`
});
```

Now `+reboot` once more and make a **typo** on purpose. For example, let's try using the `invite` command that Klasa gives us. But we will type `+onvite` so we can test the command editing. You will notice nothing happens, because we don't have any command by the name of `onvite`. However, if you edit your message and change it to `+invite` you will see the command run successfully.

## Final Note On Client Options

> There are a lot of client options that we will not cover all of them in this guide. To see all the options, please check XXX. If you need help with any of these options, you can join the discord server and ask for help there anytime.

Reviewer: ```suggestion
There are a lot of client options that we will not cover all of them in this guide. To see all the options, please check {@link KlasaClientOptions}. If you need help with any of these options, you can join the discord server and ask for help there anytime.
```
